src/linear_algebra/eigenspace.lean

```diff
@@ -51,8 +51,7 @@ variables {K R : Type v} {V M : Type w}
 
 /-- The submodule `eigenspace f μ` for a linear map `f` and a scalar `μ` consists of all vectors `x`
     such that `f x = μ • x`. (Def 5.36 of [axler2015])-/
-def eigenspace (f : End R M) (μ : R) : submodule R M :=
-(f - algebra_map R (End R M) μ).ker
+def eigenspace (f : End R M) (μ : R) : submodule R M := (f - μ • 1).ker
```

Comment:
`algebra_map R A x` is essentially a way to let the user provide a definitionally convenient version of `x • 1` - so whatever convenience that definition may permit, this removes it.

A safer bet might be to just provide a lemma `eigenspace_eq_sub_smul_one_ker` or something to obtain this definition.

Reply:
Is there a motivation here beyond "`algebra_map K (End K V)` is long, `• 1` is short and more readable"?

Reply:
I should have clarified in the PR since my motivation for this change is buried in the commit message of [this commit](https://github.com/leanprover-community/mathlib/pull/7285/commits/759a649744f7265d3b005a2b4d447545017e3a7a).

In fact I wish to remove the definitional convenience you mention. It is convenient for the current definition of `algebra` which I want to change (to allow algebras over non-assoc, non-unital rings). Much of the difficulty I encountered when hacking on the `replace_algebra_def` branch was caused by code exercising this definitional convenience. I saw an easy opportunity to reduce the surface area of that problem slightly here so I decided to propose this change.

I confess I do like that the character count shrinks but I claim I would happily make the proposal if somehow it were the opposite.

Reply:
The definitional convenience can be kept and my proposal may indeed be a bad idea but I don't (yet!) think so.

It may help if I am more explicit than my remarks about this above.

In order to be able to talk about algebras which are not-necessarily-associative and not-necessarily-unital ([once such "rings" exist](https://github.com/leanprover-community/mathlib/pull/6786)) I would like to change the definition of `algebra` from:
```lean
class algebra (R : Type u) (A : Type v) [comm_semiring R] [semiring A] extends has_scalar R A, R →+* A :=
(commutes' : ∀ r x, to_fun r * x = x * to_fun r)
(smul_def' : ∀ r x, r • x = to_fun r * x)
```
to:
```lean
class algebra (R : Type u) [semiring R] (A : Type v) [semiring A] extends semimodule R A :=
(smul_mul_assoc' : ∀ (t : R) (a b : A), (t • a) * b = t • (a * b))
(mul_smul_comm' : ∀ (t : R) (a b : A), a * (t • b) = t • (a * b))
```

I tried to do this a few weeks ago and found that it was a lot of work so I shelved the effort (I intend to try again properly in a week or two). Unsurprisingly, in the course of that effort I found that code mentioning `algebra_map` tended to need attention. I believe that the change proposed here will save me from having to provide such attention when I change the definition of `algebra`.

However I admit there is also a good case to be made for leaving `eigenspace` as `(f - algebra_map R (End R M) μ).ker` at least until I actually make the change to the `algebra` definition (especially as there is a chance I do not succeed in that effort).

As I say, I favour these changes but I think it is entirely reasonable to say that I should drop some / all of them.